File: lib/models/Chapter.dart
class Chapter {
  int? id;
  String? title;
  String? content;
  int? chapterNumber;
  DateTime? createAt;
  DateTime? updateAt;
  int? view;
  int? test; // Giá trị này sẽ không được serialize/deserialize

  Chapter(
      {this.id,
        this.title,
        this.content,
        this.chapterNumber,
        this.createAt,
        this.updateAt,
        this.view,
        this.test});

  Chapter.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    title = json['title'];
    content = json['content'];
    chapterNumber = json['chapterNumber'];
    createAt = json['createAt'] != null
        ? DateTime.parse(json['createAt'])
        : null;
    updateAt = json['updateAt'] != null
        ? DateTime.parse(json['updateAt'])
        : null;
    view = json['view'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['title'] = title;
    data['content'] = content;
    data['chapterNumber'] = chapterNumber;
    data['createAt'] = createAt?.toIso8601String();
    data['updateAt'] = updateAt?.toIso8601String();
    data['view'] = view;
    return data;
  }
}

File: src/main/java/com/example/noveltruyen/Model/Chapter.java
package com.example.noveltruyen.Model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@NoArgsConstructor
@AllArgsConstructor
@Getter @Setter
@Entity
@Table(name = "chapters")
public class Chapter {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "title", nullable = false)
    private String title;

    @Lob
    @Column(name = "content", columnDefinition = "TEXT")
    private String content;

    @Column(name = "chapter_number", nullable = false)
    private int chapterNumber;

    @Column
    @CreationTimestamp
    private LocalDateTime createAt;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "story_id")
    private Story story;

    @UpdateTimestamp
    @Column
    private LocalDateTime updateAt;

    private int view;
    @Transient
    private int test;


}